bcl/src/bc_i2c.c

```diff
@@ -299,3 +374,278 @@ bool bc_i2c_read_16b(bc_i2c_channel_t channel, uint8_t device_address, uint32_t
+    }
+
+    /* Send Slave Address and Memory Address */
+    if (!_bc_i2c_req_mem_read(I2Cx, device_address, memory_address, memory_address_length, 10, &tick_start))
+    {
+        return false;
+    }
+
+    return _bc_i2c_read(I2Cx, device_address, buffer, length, &tick_start);
+}
+
+static inline bool _bc_i2c_req_mem_write(I2C_TypeDef *I2Cx, uint8_t device_address, uint16_t memory_address, uint16_t memory_address_length, bc_tick_t timeout, bc_tick_t *tick_start)
+{
+    _bc_i2c_config(I2Cx, device_address, memory_address_length, _BC_I2C_RELOAD_MODE, _BC_I2C_GENERATE_START_WRITE);
+
+    /* Wait until TXIS flag is set */
+    if (!_bc_i2c_watch_flag(I2Cx, I2C_ISR_TXIS, RESET, timeout, tick_start))
+    {
+        return false;
+    }
+
+    /* If Memory address size is 8Bit */
+    if (memory_address_length == _BC_I2C_MEMORY_ADDRESS_SIZE_8BIT)
+    {
+        /* Send Memory Address */
+        I2Cx->TXDR = memory_address & 0xff;
+    }
+    /* If Memory address size is 16Bit */
+    else
+    {
+        /* Send MSB of Memory Address */
+        I2Cx->TXDR = (memory_address >> 8) & 0xff;
+
+        /* Wait until TXIS flag is set */
+        if (!_bc_i2c_watch_flag(I2Cx, I2C_ISR_TXIS, RESET, timeout, tick_start))
+        {
+            return false;
+        }
+
+        /* Send LSB of Memory Address */
+        I2Cx->TXDR = memory_address & 0xff;
+    }
+
+    /* Wait until TCR flag is set */
+    if (!_bc_i2c_watch_flag(I2Cx, I2C_ISR_TCR, RESET, timeout, tick_start))
+    {
+        return false;
+    }
+
+    return true;
+}
+
+static inline bool _bc_i2c_req_mem_read(I2C_TypeDef *I2Cx, uint8_t device_address, uint16_t memory_address, uint16_t memory_address_length, bc_tick_t timeout, bc_tick_t *tick_start)
+{
+    _bc_i2c_config(I2Cx, device_address, memory_address_length, _BC_I2C_SOFTEND_MODE, _BC_I2C_GENERATE_START_WRITE);
+
+    /* Wait until TXIS flag is set */
+    if (_bc_i2c_watch_flag(I2Cx, I2C_ISR_TXIS, RESET, timeout, tick_start) != true)
+    {
+        return false;
+    }
+
+    /* If Memory address size is 8Bit */
+    if (memory_address_length == _BC_I2C_MEMORY_ADDRESS_SIZE_8BIT)
+    {
+        /* Send Memory Address */
+        I2Cx->TXDR = memory_address & 0xff;
+    }
+    /* If Memory address size is 16Bit */
+    else
+    {
+        /* Send MSB of Memory Address */
+        I2Cx->TXDR = (memory_address >> 8) & 0xff;
+
+        /* Wait until TXIS flag is set */
+        if (_bc_i2c_watch_flag(I2Cx, I2C_ISR_TXIS, RESET, timeout, tick_start) != true)
+        {
+            return false;
+        }
+
+        /* Send LSB of Memory Address */
+        I2Cx->TXDR = memory_address & 0xff;
+    }
+
+    /* Wait until TC flag is set */
+    if (_bc_i2c_watch_flag(I2Cx, I2C_ISR_TC, RESET, timeout, tick_start) != true)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+static void _bc_i2c_config(I2C_TypeDef *I2Cx, uint8_t device_address, uint8_t length, uint32_t mode, uint32_t Request)
+{
+    uint32_t reg = 0U;
```

Comment:
Why do we initialize the reg variable when it gets immediately overwritten in the next statement?